src/canvas/canvasRenderer.js

```diff
@@ -0,0 +1,166 @@
+  m_this.ccl_layer = arg.layer;
+
+  /// TODO: Move this API to the base class
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Return width of the renderer
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.width = function () {
+    return m_width;
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Return height of the renderer
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.height = function () {
+    return m_height;
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Get API used by the renderer
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.api = function () {
+    return 'canvas';
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Initialize
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this._init = function () {
+    if (m_this.initialized()) {
+      return m_this;
+    }
+
+    s_init.call(m_this);
+
+    var canvas = $(document.createElement('canvas'));
+    m_this.context2d = canvas[0].getContext('2d');
+
+    canvas.attr('class', 'canvas-canvas');
+    $(m_this.layer().node().get(0)).append(canvas);
+
+    m_this.canvas(canvas);
+    /* Initialize the size of the renderer */
+    var map = m_this.layer().map(),
+        mapSize = map.size();
+    m_this._resize(0, 0, mapSize.width, mapSize.height);
+
+    return m_this;
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Handle resize event
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this._resize = function (x, y, w, h) {
+    m_width = w;
+    m_height = h;
+    m_this.canvas().attr('width', w);
+    m_this.canvas().attr('height', h);
+    m_this._render();
+
+    return m_this;
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Render
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this._render = function () {
+    if (m_renderAnimFrameRef === null) {
+      m_renderAnimFrameRef = window.requestAnimationFrame(function () {
+        m_renderAnimFrameRef = null;
+        m_this.context2d.fillStyle = '#33FFFF';
+        m_this.context2d.fillRect(300, 200, 450, 175);
+
+        var layer = m_this.layer(),
+            map = layer.map(),
+            camera = map.camera(),
+            viewport = camera._viewport;
+        // Clear the canvas.
+        m_this.context2d.setTransform(1, 0, 0, 1, 0, 0);
+        m_this.context2d.clearRect(0, 0, viewport.width, viewport.height);
+
+        var features = layer.features();
+        // loop ?
+        features[0]._renderImageQuads(m_this.context2d, map);
```

Comment:
Eventually, we'd like a renderer to support different layer types with different features.  It might be better to change this to something like:
```
for (var i = 0; i < features.length; i += 1) {
  feature[i]._renderOnCanvas(m_this.context2d, map);
}
```
and place the call to `_renderImageQuads` in the quadFeature class.